### ISSUE TITLE
mdAutocomplete not passing Accessibility Test

### ISSUE THREAD

**Issue by ByzanTine** [angular/material#6042]
![image](https://cloud.githubusercontent.com/assets/5234867/11545654/4c9f7c1a-98fd-11e5-954b-b2aab5942d3b.png)
There are two errors.
Elements with ARIA roles must ensure required owned elements are present.
This element has an unsupported ARIA attribute.
Run the Chrome Accessibility Tool inside Audit in Chrome Developer Tool will replay this issue.

**Comment by marcysutton** [comment]
For the owned elements error:
You could try moving `aria-owns="md-autocomplete-suggestions"` from `input` to mdAutocomplete (which has `role="listbox"`) and changing `role="button"` to `role="option"` on each virtual repeat list item.

For the unsupported ARIA attribute error:
Try moving `aria-autocomplete="listbox"` to mdAutocomplete, which has the `listbox` role. http://www.w3.org/TR/wai-aria/states_and_properties#aria-autocomplete

**Comment by marcysutton** [comment]
Here is a combobox demo: http://oaa-accessibility.org/example/10/

**Comment by jelbourn** [comment]
@marcysutton @ByzanTine: does the autocomplete actually have an usability problems in our supported browser + screen reader combinations? The audit from the tool is useful, but it's possible everything is working fine as is, in which case I would put a lower priority on this.

**Comment by marcysutton** [comment]
In Voiceover and Safari, the field is announced as "search text field", but not listbox...so it's hard to tell that it's anything other than a plain text field. But I could see this kind of issue being a lower priority, since it is currently operable.

In JAWS 16 and IE, we have a bigger problem: although the widget is announced correctly as a listbox, the options and screen reader messages are all announced literally as "left brace left brace message right brace right brace", so there something wrong with the interpolation. This makes the widget completely unusable. If someone can check it in JAWS 17, that would be good.

NVDA and Firefox seems to work okay, but I'd be curious to hear from a proficient screen reader user. On touch devices with a screen reader running, the autocomplete options are pretty unusable...but I suspect most autocompletes would suffer from the same problem. 

In short: with the exception of JAWS incorrectly reading messages, any changes could come after actual user testing.

**Comment by biomassives** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  acmeideal@gmail.com
    Domain biomassiv.es has exceeded the max emails per hour (501/500 (100%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext8.iad.github.net ([192.30.252.199]:43267 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by ps5.websitehostserver.net with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.86_1)
	(envelope-from <noreply@github.com>)
	id 1avsbV-000KoV-H8
	for greg@biomassiv.es; Thu, 28 Apr 2016 15:32:05 -0500
Date: Thu, 28 Apr 2016 13:26:14 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1461875174;
	bh=MoAnj795TTYgyFc8IFWJRLdllCsMaMO3KHUBy1HrmqE=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=2f5EjBiRGEhIQ91VDOqz+KpBG5p5CaRgdmi8N5XVjPYMe/HtcPXu3+aO4EjufLyUu
	 HgQj+QApJoqq9CVEbJPCjhACKtqKjyChFqxhu3pxkVGswRTu/Zh05CUjGd7oowtZpV
	 CeIS2Hgmz0aRGCtNduWKTrE4+NMWg9LjUL3jjvnY=
